### PR TITLE
add GET products by Id

### PR DESCRIPTION
**Pull request by DerEns86**
-Create a GET "/api/products/{id}" endpoint to fetch the product specified by the id.
 -Display all informations given from the backend on a extra Route for each product "/products/:id

### CODE REVIEW

File: backend/src/test/java/de/neuefische/backend/controller/ProductsControllerTest.java
Comment (gcode-de) [inline backend/src/test/java/de/neuefische/backend/controller/ProductsControllerTest.java:1]: You should also test edge cases e.g. product not found.

File: backend/src/main/java/de/neuefische/backend/service/ProductService.java
Comment (gcode-de) [inline backend/src/main/java/de/neuefische/backend/service/ProductService.java:1]: You should also test edge cases e.g. product not found.

File: backend/src/test/java/de/neuefische/backend/service/ProductServiceTest.java
Comment (gcode-de) [inline backend/src/test/java/de/neuefische/backend/service/ProductServiceTest.java:1]: Same here: You should also test edge cases e.g. product not found.

File: frontend/src/App.css [hunk 1]
```diff
@@ -0,0 +1,28 @@
+
+h3{
+    text-align: center;
+    padding: 8px 16px 8px 16px;
```
Comment (gcode-de) [inline frontend/src/App.css:4]:
```suggestion
    padding: 8px 16px;
```

Comment (gcode-de) [reply frontend/src/App.css:4]: you can simplify like that

File: frontend/src/components/ProductDetail.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    const [product, setProduct] = useState<Product>()
+
+    function getProduct() {
+        axios.get(`/api/products/${id}`)
+            .then(response => {
+                setProduct(response.data)
+                console.log('Product fetched')
+                console.log(response.data)
+            })
+            .catch(error => {
```
Comment (gcode-de) [inline frontend/src/components/ProductDetail.tsx:21]: It's best practice to remove console logs in production to avoid revealing too much information about how the system works in the background.
Console.error in line 22 should be finde though.

**Review by gcode-de — APPROVED** [review]
nicely done!
I've added some small suggestions.
